systems/registry/node/pkg/server/node.go

```diff
@@ -198,51 +174,45 @@ func (n *NodeServer) GetNodes(ctx context.Context, req *pb.GetNodesRequest) (*pb
 	return n.getAllNodes(ctx, req)
 }
 
-func (n *NodeServer) UpdateNodeState(ctx context.Context, req *pb.UpdateNodeStateRequest) (*pb.UpdateNodeResponse, error) {
+func (n *NodeServer) UpdateNodeStatus(ctx context.Context, req *pb.UpdateNodeStateRequest) (*pb.UpdateNodeResponse, error) {
 	log.Infof("Updating node state  %v", req.GetNodeId())
 
-	dbState := db.ParseNodeState(req.State)
+	dbNodeState := db.ParseNodeState(req.State)
+	dbConnState := db.ParseConnectivityState(req.Connectivity)
 
 	nodeId, err := ukama.ValidateNodeId(req.GetNodeId())
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	nodeUpdates := &db.Node{
-		Id:    nodeId.StringLowercase(),
-		State: dbState,
+	nodeUpdates := &db.NodeStatus{
+		NodeId: nodeId.StringLowercase(),
 	}
 
-	err = n.nodeRepo.Update(nodeUpdates, nil)
-	if err != nil {
-		log.Error("error updating the node state, ", err.Error())
-
-		return nil, grpc.SqlErrorToGrpc(err, "node")
-	}
+	pbStatus := &pb.NodeStatus{}
 
-	resp := &pb.UpdateNodeResponse{
-		Node: &pb.Node{
-			Id:    req.GetNodeId(),
-			State: req.State,
-		},
+	if req.State != "" {
+		nodeUpdates.State = dbNodeState
+		pbStatus.State = dbNodeState.String()
 	}
-	und, err := n.nodeRepo.Get(nodeId)
-	if err != nil {
-		log.Error("error getting the node, ", err.Error())
 
-		return resp, nil
+	if req.Connectivity != "" {
+		nodeUpdates.Conn = dbConnState
+		pbStatus.Connectivity = dbConnState.String()
 	}
 
-	route := n.baseRoutingKey.SetAction("updatestate").SetObject("node").MustBuild()
+	err = n.nodeStatusRepo.Update(nodeUpdates)
+	if err != nil {
+		log.Error("error updating the node state, ", err.Error())
 
-	evt := &epb.NodeStateUpdatedEvent{
-		NodeId: nodeUpdates.Id,
-		State:  nodeUpdates.State.String(),
+		return nil, grpc.SqlErrorToGrpc(err, "node")
 	}
 
-	err = n.msgbus.PublishRequest(route, evt)
+	und, err := n.nodeRepo.Get(nodeId)
 	if err != nil {
-		log.Errorf("Failed to publish message %+v with key %+v. Errors %s", evt, route, err.Error())
+		log.Error("error updating the node state, ", err.Error())
```

Comment:
I think  if we were able to successfully proceed to NodeStateUpdate(), but are failing to get back the corresponding node with NodeRepo.Get() after that, then this is even a more severe error than failing to update the node's state.  This probably means that the Node DB and the NodeState DB are in an inconsistent state in regard to each other.

Reply:
Failed to get could be due to number of factors whicj are very unlikely to happen like db unreachable (nw failure) or that something that kind of stuff. I dont think adding err clause here was meant for like,issing node id or sometink like that.